java/spark/src/main/java/com/lancedb/lance/spark/LanceIdentifier.java

```diff
@@ -14,13 +14,85 @@
 package com.lancedb.lance.spark;
 
 import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * LanceIdentifier is a custom implementation of {@link Identifier} for Lance. It contains the
+ * dataset URI and the namespace. The namespace is an array of strings, which contains the namespace
+ * of the dataset and the options. The options are key-value pairs, which are separated by "#####".
+ */
 public class LanceIdentifier implements Identifier {
-  private final String[] namespace = new String[] {"default"};
+  public static final String SEPARATOR = "#####";
```

Comment:
Is `####` a customary separator or delimiter in spark ecosystem? We may need some comments or remarks to describe its purpose?